client/src/components/graph/graph.tsx

```diff
@@ -1008,7 +1025,51 @@ class Graph extends React.Component<GraphProps, GraphState> {
       this.isDownloadingImage = true;
 
       const graph = regl._gl.canvas;
-      const imageURI = await toPng(graph as HTMLCanvasElement, {
+
+      const offscreenCanvas = document.createElement("canvas");
+      offscreenCanvas.width = width;
+      offscreenCanvas.height = height;
+      const legendNodes =
+        document.getElementById("continuous_legend")?.childNodes;
+      const ctx = offscreenCanvas.getContext("2d");
+      ctx?.drawImage(graph, 0, 0);
+
+      legendNodes?.forEach(async (node) => {
+        if (node.nodeName === "CANVAS") {
+          // get style from node
+          const style = window.getComputedStyle(node as HTMLCanvasElement);
+          const nodeWidth = parseInt(style.width, 10);
+          const nodeHeight = parseInt(style.height, 10);
+          const nodeLeft = parseInt(style.left, 10);
+          const nodeTop = parseInt(style.top, 10);
+          ctx?.drawImage(
+            node as HTMLCanvasElement,
+            100 + nodeLeft,
+            100 + nodeTop,
+            nodeWidth,
+            nodeHeight
+          );
+        } else if (node.nodeName === "svg" || node.nodeName === "SVG") {
+          const svgNode = node as SVGSVGElement;
+          const xml = new XMLSerializer().serializeToString(svgNode);
+          const svg64 = btoa(unescape(encodeURIComponent(xml)));
+          const b64Start = "data:image/svg+xml;base64,";
+          const image64 = b64Start + svg64;
+          const img = new Image();
+          img.onload = () => {
+            ctx?.drawImage(img, 100, 100);
```

Comment:
is the 100 here linked to the offset `100` above? `100 + nodeLeft`

If so, let's use a constant to link them instead